keras/src/models/functional.py

```diff
@@ -181,6 +181,9 @@ def compute_output_spec(self, inputs, training=None, mask=None):
         # From Function
         return super().compute_output_spec(inputs)
 
+    def compute_output_shape(self, input_shape):
+        return self.output_shape
```

Comment:
Got it.
I realized that it would be hard work to fully support `compute_output_shape` in the `Functional` model, because we would need to add that function to both `Function` and `Operation`.

I have used a workaround that implements `compute_output_shape` by utilizing `compute_output_spec`.
This should work well and require only small changes to the codebase.